packages/dds/tree/src/shared-tree-core/branch.ts

```diff
@@ -82,6 +83,8 @@ export function getChangeReplaceType(
 	// B' is removed and replaced by B because both have the same revision.
 	if (
 		change.removedCommits.length === 1 &&
+		change.removedCommits[0] !== undefined &&
```

Comment:
Checking change.removeCommits[0] !== undefined should be an assert, not a conditional (we already checked that it has at least one element, above on line 73. We know the length is not 0).

Same for `change.newCommits`, although for a different reason - a replace operation always has new commits.